README.rst

```diff
@@ -88,105 +87,62 @@ Documentation
 
-      >>> # continuing from above
-      >>> mt_dev = datasets.TranslationDataset(
-      ...     path='data/mt/newstest2014', exts=('.en', '.de'),
-      ...     fields=(src, trg))
-      >>> src.build_vocab(mt_train, max_size=80000)
-      >>> trg.build_vocab(mt_train, max_size=40000)
-      >>> # mt_dev shares the fields, so it shares their vocab objects
-      >>>
-      >>> train_iter = data.BucketIterator(
-      ...     dataset=mt_train, batch_size=32,
-      ...     sort_key=lambda x: data.interleave_keys(len(x.src), len(x.trg)))
-      >>> # usage
-      >>> next(iter(train_iter))
-      <data.Batch(batch_size=32, src=[LongTensor (32, 25)], trg=[LongTensor (32, 28)])>
-
-* Wrapper for dataset splits (train, validation, test):
+For example, to access the raw text from the AG_NEWS dataset:
 
   .. code-block:: python
 
-      >>> TEXT = data.Field()
-      >>> LABELS = data.Field()
-      >>>
-      >>> train, val, test = data.TabularDataset.splits(
-      ...     path='/data/pos_wsj/pos_wsj', train='_train.tsv',
-      ...     validation='_dev.tsv', test='_test.tsv', format='tsv',
-      ...     fields=[('text', TEXT), ('labels', LABELS)])
-      >>>
-      >>> train_iter, val_iter, test_iter = data.BucketIterator.splits(
-      ...     (train, val, test), batch_sizes=(16, 256, 256),
-      >>>     sort_key=lambda x: len(x.text), device=0)
-      >>>
-      >>> TEXT.build_vocab(train)
-      >>> LABELS.build_vocab(train)
+      >>> from torchtext.datasets import AG_NEWS
+      >>> train_iter = AG_NEWS(split='train')
+      >>> next(train_iter)
+      >>> # Or iterate with for loop
+      >>> for (label, line) in train_iter:
+      >>>     print(label, line)
+      >>> # Or send to DataLoader
+      >>> from torch.utils.data import DataLoader
+      >>> train_iter = AG_NEWS(split='train')
+      >>> dataloader = DataLoader(train_iter, batch_size=8, shuffle=False)
 
-Datasets
-========
+A tutorial for the end-to-end text classification workflow can be found in `PyTorch tutorial <https://pytorch.org/tutorials/beginner/text_sentiment_ngrams_tutorial.html>`_
 
-The datasets module currently contains:
+[Prototype] Experimental Code
+=============================
 
-* Sentiment analysis: SST and IMDb
-* Question classification: TREC
-* Entailment: SNLI, MultiNLI
-* Language modeling: abstract class + WikiText-2, WikiText103, PennTreebank
-* Machine translation: abstract class + Multi30k, IWSLT, WMT14
-* Sequence tagging (e.g. POS/NER): abstract class + UDPOS, CoNLL2000Chunking
-* Question answering: 20 QA bAbI tasks
-* Text classification: AG_NEWS, SogouNews, DBpedia, YelpReviewPolarity, YelpReviewFull, YahooAnswers, AmazonReviewPolarity, AmazonReviewFull
+We have re-written several building blocks under ``torchtext.experimental``:
 
-Others are planned or a work in progress:
+* Transforms
+* Vocabulary
+* Vectors
 
```

Comment:
Should we provide short note or one liner description of each of "Transforms", "Vocabulary" and "Vectors". Or at-least a link where users can learn more about it?  Since these are prototype features where we probably want some feedback, users might want to learn about them quickly through somewhere?

Reply:
Sure. Let me add those links and some descriptions.